### PR TITLE
Polish Hive Docs

### PR DESCRIPTION
**Pull request by dimaMachina**
Restructure and polishing Nextra 4 migration to simplify things

image

### CODE REVIEW

File: packages/web/app/package.json [hunk 1]
```diff
@@ -6,8 +6,6 @@
     "build": "tsx ../../../scripts/runify.ts src/server/index.ts && vite build --outDir dist/client",
     "build-storybook": "storybook build",
     "dev": "node --env-file=.env --watch-path src/server --import tsx src/server/index.ts --dev",
-    "generate-changelog": "node ../../../scripts/generate-changelog.js",
-    "postinstall": "pnpm generate-changelog",
```
Comment (dimaMachina) [inline packages/web/app/package.json:10]: moved inside `packages/web/docs/src/app/feed.xml/route.ts`

File: .gitignore [hunk 2]
```diff
@@ -127,8 +127,6 @@ packages/web/app/next.config.mjs
 packages/web/app/environment-*.mjs
 packages/web/app/src/gql/*.ts
 packages/web/app/src/gql/*.json
-# Changelog
-packages/web/app/src/components/ui/changelog/generated-changelog.ts
```
Comment (dimaMachina) [inline .gitignore:131]: I think it's better to save to git, to actually see what's happens with content of this file

File: packages/web/docs/next.config.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ export default withGuildDocs({
       treeShaking: true,
     },
   },
+  nextraConfig: {
+    contentDirBasePath: '/docs',
```
Comment (dimaMachina) [inline packages/web/docs/next.config.js:15]: @hasparus this is magic trick

File: packages/web/docs/src/app/docs/[[...mdxPath]]/page.tsx [hunk 4]
```diff
@@ -3,36 +3,28 @@ import { generateStaticParamsFor, importPage } from 'nextra/pages';
...
 export async function generateMetadata(props: PageProps<'...mdxPath'>) {
-  const params = await props.params;
-  const { metadata } = await importPage(['docs', ...(params.mdxPath || [])]);
-  return metadata;
+  const { mdxPath } = await props.params;
+  const { metadata } = await importPage(mdxPath);
+  return {
+    ...metadata,
+    ...(mdxPath?.[0] === 'gateway' && {
+      title: { absolute: `${metadata.title} | Hive Gateway` },
```
Comment (dimaMachina) [inline packages/web/docs/src/app/docs/[[...mdxPath]]/page.tsx:14]: moved logic from `packages/web/docs/src/app/dynamic-meta-tags.tsx`

File: packages/web/docs/src/app/federation/page.mdx [hunk 5]
```diff
@@ -1,9 +0,0 @@
----
-title: What is GraphQL Federation?
-description:
-  Discover what GraphQL Federation is, how it unifies multiple APIs into a Supergraph, its core
-  benefits, and the building blocks like subgraphs, schema composition and gateway.
-ogImage: /federation-og-image.png
----
-
-export { FederationPage as default } from '../../components/federation-page'
```
Comment (dimaMachina) [inline packages/web/docs/src/app/federation/page.mdx:9]: we don't need here `page.mdx`, should be `page.tsx`

File: packages/web/docs/src/app/oss-friends/page.mdx [hunk 6]
```diff
@@ -1,8 +0,0 @@
----
-title: Our Open Source Friends
-description: We love open source. Meet our friends who share the same passion.
----
-
-import { OSSFriendsPage } from '../../components/oss-friends'
-
-<OSSFriendsPage />
```
Comment (dimaMachina) [inline packages/web/docs/src/app/oss-friends/page.mdx:8]: same for this route, no need `page.mdx`

File: packages/web/docs/src/components/product-updates.tsx [hunk 7]
```diff
@@ -42,81 +42,28 @@ function ProductUpdateTeaser(props: Changelog): ReactElement {
...
-
-  return productUpdatesFolder
-    .slice(1) // cut `_meta.ts` which always comes first
+  return pageMap
     .map(item => {
-      if (!('children' in item) && 'route' in item) {
-        const frontMatter = (item as any).frontMatter as {
-          title: string;
-          date?: Date;
-          description: string;
```
Comment (dimaMachina) [inline packages/web/docs/src/components/product-updates.tsx:78]: @hasparus see how I simplified it

Comment (hasparus) [reply packages/web/docs/src/components/product-updates.tsx:78]: Okay, this is soo nice.

File: packages/web/docs/src/content/self-hosting/changelog.mdx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+# Self-Hosting Changelog
+
+Here you can find all the recent improvements and updates about the self-hosting images. The Docker
+images published to the GitHub registry using the same tags. For a high-level overview of
+improvements please refer to [our product updates](/product-updates).
+
+export const source = await fs.readFile(path.resolve('../../../deployment/CHANGELOG.md'), 'utf8')
+
+<MDXRemote compiledSource={await compileMdx(source.replace('# hive', ''))} />
```
Comment (hasparus) [inline packages/web/docs/src/content/self-hosting/changelog.mdx:19]: Should we switch to something like

```suggestion
<MDXRemote compiledSource={await compileMdx(source.replace(/#\s+.*\n/, ''))} />
```

so it still works if someone oblivious uppercases the "Hive" in that heading or changes the word?

**Review by hasparus — APPROVED** [review]

**Pull request merged**